DynamicActiveQuery.php

```diff
@@ -209,11 +209,11 @@ public function createCommand($db = null)
         }
 
         $dynamicColumn = $modelClass::dynamicColumn();
-        $callback = function ($matches) use (&$params, $dynamicColumn) {
+        $callback = function ($matches) use (&$params, $dynamicColumn, $modelClass) {
             $type = !empty($matches[3]) ? $matches[3] : 'CHAR';
             $sql = $dynamicColumn;
             foreach (explode('.', $matches[2]) as $column) {
-                $placeholder = DynamicActiveRecord::placeholder();
+                $placeholder = $modelClass::getDynamicEncoder()->placeholder();
```

Comment:
or fetch the Encoder object in the enclosing scope and `use` that instead of `$modelClass`?